data_hub_api/docmaps/v2/sql/docmaps_index.sql

```diff
@@ -302,11 +302,42 @@ t_rp_meca_path_update AS (
   WHERE rn=1
 ),
 
+t_manual_preprint_match_for_published_date AS (
+  SELECT
+    * EXCEPT(rn),
+  FROM
+  (
+    SELECT 
+      *,
+      ROW_NUMBER() OVER(PARTITION BY long_manuscript_identifier ORDER BY imported_timestamp DESC) AS rn
+    FROM `elife-data-pipeline.prod.unmatched_manuscripts`
+    WHERE preprint_published_at_date IS NOT NULL 
+      AND preprint_published_at_date != ''
+  )
+  WHERE rn=1
+),
+
+t_europepmc_preprint_publication_date AS (
+  SELECT
+    response.doi,
+    response.firstPublicationDate,
+    url_list.url
+  FROM `elife-data-pipeline.prod.v_latest_europepmc_preprint_servers_response`  AS response
+  LEFT JOIN UNNEST(fullTextUrlList.fullTextUrl) AS url_list
```

Comment:
I would say after unnesting, this no longer a list but a url, or some form of a wrapped url? Maybe could call it `full_text_url` as a distinction?
I guess in theory we could have multiple full text URLs, and even one was say "arxiv.org" and the other not, then we would still use it?

There are examples:

```sql
SELECT doi, fullTextUrlList.fullTextUrl
FROM `elife-data-pipeline.prod.v_latest_europepmc_preprint_servers_response`
WHERE ARRAY_LENGTH(fullTextUrlList.fullTextUrl) > 1
  AND doi LIKE '%osf.io%'
```



Row | doi | fullTextUrl.availability | fullTextUrl.availabilityCode | fullTextUrl.documentStyle | fullTextUrl.site | fullTextUrl.url |  
-- | -- | -- | -- | -- | -- | -- | --
1 | 10.31234/osf.io/b29cs | Free | F | pdf | Unpaywall | https://psyarxiv.com/b29cs/download |  
  |   | Free | F | doi | DOI | https://doi.org/10.31234/osf.io/b29cs |  
2 | 10.31234/osf.io/98hw3 | null | F | pdf | Unpaywall | https://psyarxiv.com/98hw3/download |  
  |   | Free | F | doi | DOI | https://doi.org/10.31234/osf.io/98hw3 |  
3 | 10.31234/osf.io/rjt7n | null | F | pdf | Unpaywall | https://psyarxiv.com/rjt7n/download |  
  |   | Free | F | doi | DOI | https://doi.org/10.31234/osf.io/rjt7n



But using the DOI, if we can, seems slightly safer, as there may not be a full text URL or the full text URL could also point to something not containing osf.io in the URL?